doc/design/cluster_train/submit-job.md

```diff
@@ -0,0 +1,133 @@
+
+### Runtime Docker Image
+
+  The trainer package which user upload and some Python dependencies are packaged into a runtime Docker image based on base Docker image.
+
+- Handle Python Dependencies
+
+  You need to provide requirements.txt file in your `trainer-package` folder. Example:
+
+  ```txt
+  pillow
+  protobuf==3.1.0
+  ```
+  More [details](https://pip.readthedocs.io/en/1.1/requirements.html) about requirements, an example project looks like:
+  ```bash
+    paddle_example
+      |-quick_start
+        |-trainer.py
+        |-dataset.py
+        |-requirements.txt
+  ```
+
+## Submit Distributed Training Job With Python Code
+<img src="./src/submit-job-python.png" width="800">
+
+- `paddle.job.dist_train()` will call the Job Server API `/v1/packages` to upload the trainer package and save them on CephFS, and then call `/v1/trainer/job` to submit the PaddlePaddle distributed job.
+- `/v1/trainer/job` will start a building job for preparing the runtime Docker image. When the building job is finished, Job Server will submit the PaddlePaddle distributed job to Kubernetes.
+- *NOTE*: For the first version, we will not prepare the runtime docker image, instead, we will mount the trainer package in a temporary folder into the base docker image. We will not support custom Python dependencies in the first version as well.
+
+You can call `paddle.job.dist_train` and provide distributed training configuration as the parameters:
+```python
+paddle.job.dist_train(
+    trainer=paddle.trainer.SGD(...,
+                              paddle.updater.Adam(...)),
+    reader=reader,
+    paddle_job=PaddleJob(
+      pserver_bucket="stander",
+      base_image="yancey1989/paddle-cloud",
+      job_name="paddle-job",
+      namespace="paddle-cloud",
+      cpu_num=3,
+      memory="1G"
+      trainer_package="/example/word2vec",
+      entry_point="python %s" % __file__))
+```
+
+The pseudo code of `paddle.job.dist_train` is as follows:
+```python
+  def dist_train(trainer, reader, num_passes=1, event_handler=None, feeding=None, paddle_job=None):
+    # if the code is running on cloud, set PADDLE_ON_CLOUD=YES
+    if os.getenv("PADDLE_ON_CLOUD", "NO") == "NO":
+      #submit the paddle job
+      paddle_job.submit()
+    else:
+      #start the training
+      trainer.train(reader, num_passes, event_handler, feeding)
+```
+
+parameter | required | default | explain
+  --- | --- | --- | ---
+job_name|YES||you should special a unique job name which in a namespace
+entry_point|YES|| entry point for startup trainer process
+trainer_package|YES|| trainer package file path, you can special a cloud path with `pfs://home/paddle` or a normal path with `/home/paddle`
+base_image|YES||PaddlePaddle production Docker image
+memory|YES|| memory for the trainers totally used
+cpu_num|YES|1| CPU count for the trainers totally used
```

Comment:
> 光光cpu_num这个参数好像不够：
比如说这里写了5，是说起1个pod有5个cpu还是说起5个pod每个pod有1个cpu

CPU的total trainer可以和GPU一样，由我们的调度系统来优化，如果觉得比较复杂，我们可以在这一版分开指定。

在CPU模式下，如果指定了total trainer，还是指定每个trainer的CPU数量比较好理解一些，否则的话还要根据total trainer 和total CPU做除法用户才能知道每个trainer的CPU数量, 这样 CPU模式下需要配置`trainers`，`cpu_num`两个参数，GPU模式下需要配置`total_gpu`一个参数

- CPU Mode
    - You *must* specify `trainers`, the trainer count.
    - You *must* specify `cpu_num`, the CPU count for each trainer
- GPU mode
    - You *must* specify `total_gpu`, the GPU count fo all trainers totally used.
    - You *haven't* specify `trainers`, this will be allocated by PaddlePaddle Job.

也请 @typhoonzero 关注下这个讨论。